### PR TITLE
refact(pg): ro_pool, gc and queue limit

### PR DESCRIPTION
**Pull request by antmat**

### CODE REVIEW

**Review by 3Hren — APPROVED** [review]
Whatever.

File: metrics/src/sender/postgres.cpp [hunk 1]
```diff
@@ -71,6 +71,7 @@ auto pg_sender_t::on_send_timer(const std::error_code& ec) -> void {
 
 auto pg_sender_t::on_gc_timer(const std::error_code& ec) -> void {
     if(!ec) {
+        COCAINE_LOG_DEBUG(logger, "running gc for metrics");
```
Comment (3Hren) [inline metrics/src/sender/postgres.cpp:76]: Either use capital letters for abbreviations or expand them: "GC" or "garbage collector".

File: metrics/src/sender/postgres.cpp [hunk 2]
```diff
@@ -86,21 +87,25 @@ auto pg_sender_t::on_gc_timer(const std::error_code& ec) -> void {
             gc_timer.async_wait(std::bind(&pg_sender_t::on_gc_timer, this, std::placeholders::_1));
         });
     } else {
-        COCAINE_LOG_WARNING(logger, "sender timer was cancelled");
+        COCAINE_LOG_WARNING(logger, "gc timer was cancelled");
```
Comment (3Hren) [inline metrics/src/sender/postgres.cpp]: Here too. Why warning BTW?

File: metrics/src/sender/postgres.cpp [hunk 3]
```diff
@@ -86,21 +87,25 @@ auto pg_sender_t::on_gc_timer(const std::error_code& ec) -> void {
...
     }
 }
 
 
 auto pg_sender_t::send(dynamic_t data) -> void {
     pool->execute([=](pqxx::connection_base& connection){
         try {
+            struct timeval tp;
+            gettimeofday(&tp, NULL);
+            std::string now = std::to_string((tp.tv_sec * 1000.0 + tp.tv_usec / 1000.0) / 1000.0);
```
Comment (3Hren) [inline metrics/src/sender/postgres.cpp]: `const auto`.

File: metrics/src/sender/postgres.cpp [hunk 4]
```diff
@@ -86,21 +87,25 @@ auto pg_sender_t::on_gc_timer(const std::error_code& ec) -> void {
...
+        COCAINE_LOG_WARNING(logger, "gc timer was cancelled");
     }
 }
 
 
 auto pg_sender_t::send(dynamic_t data) -> void {
     pool->execute([=](pqxx::connection_base& connection){
         try {
+            struct timeval tp;
+            gettimeofday(&tp, NULL);
```
Comment (3Hren) [inline metrics/src/sender/postgres.cpp]: I don't see `sys/time.h` include for this.

Comment (3Hren) [reply metrics/src/sender/postgres.cpp]: `NULL`, really? Where are error handling code BTW?

File: postgres/src/pool.cpp [hunk 5]
```diff
@@ -28,9 +31,15 @@ pool_t::~pool_t() {
 
 void
 pool_t::execute(std::function<void(pqxx::connection_base&)> function) {
```
Comment (3Hren) [inline postgres/src/pool.cpp:33]: Are the calling this method code REALLY ready for catching newly introduced exception?

File: metrics/src/sender/postgres.cpp [hunk 6]
```diff
@@ -86,21 +87,25 @@ auto pg_sender_t::on_gc_timer(const std::error_code& ec) -> void {
...
-        COCAINE_LOG_WARNING(logger, "sender timer was cancelled");
+        COCAINE_LOG_WARNING(logger, "gc timer was cancelled");
     }
 }
 
 
 auto pg_sender_t::send(dynamic_t data) -> void {
     pool->execute([=](pqxx::connection_base& connection){
         try {
+            struct timeval tp;
```
Comment (3Hren) [inline metrics/src/sender/postgres.cpp]: What does `tp` mean?

File: postgres/src/pool.cpp (hunk 5 above)
Comment (antmat) [reply postgres/src/pool.cpp:33]: I hope so

**Pull request merged**